circuit/src/ecc/generic/mod.rs

```diff
@@ -1,2 +1,1798 @@
 /// Native implementation
 pub mod native;
+
+use crate::{
+	gadgets::{
+		bits2integer::{Bits2IntegerChipset, Bits2IntegerChipsetConfig},
+		bits2num::Bits2NumChip,
+		main::{MainConfig, SelectChipset},
+	},
+	integer::{
+		AssignedInteger, IntegerAddChip, IntegerDivChip, IntegerMulChip, IntegerReduceChip,
+		IntegerSubChip,
+	},
+	params::rns::RnsParams,
+	utils::{assigned_as_bool, be_assigned_bits_to_usize},
+	Chip, Chipset, CommonConfig, FieldExt,
+};
+use halo2::{
+	circuit::{AssignedCell, Layouter},
+	halo2curves::CurveAffine,
+	plonk::{Error, Selector},
+};
+
+/// Structure for the AssignedPoint.
+#[derive(Clone, Debug)]
+pub struct AssignedPoint<
+	C: CurveAffine,
+	N: FieldExt,
+	const NUM_LIMBS: usize,
+	const NUM_BITS: usize,
+	P,
+> where
+	P: RnsParams<C::Base, N, NUM_LIMBS, NUM_BITS> + RnsParams<C::ScalarExt, N, NUM_LIMBS, NUM_BITS>,
+	<C as CurveAffine>::Base: FieldExt,
+	<C as CurveAffine>::ScalarExt: FieldExt,
+{
+	// x coordinate of the point
+	pub(crate) x: AssignedInteger<C::Base, N, NUM_LIMBS, NUM_BITS, P>,
+	// y coordinate of the point
+	pub(crate) y: AssignedInteger<C::Base, N, NUM_LIMBS, NUM_BITS, P>,
+}
+
+impl<C: CurveAffine, N: FieldExt, const NUM_LIMBS: usize, const NUM_BITS: usize, P>
+	AssignedPoint<C, N, NUM_LIMBS, NUM_BITS, P>
+where
+	P: RnsParams<C::Base, N, NUM_LIMBS, NUM_BITS> + RnsParams<C::ScalarExt, N, NUM_LIMBS, NUM_BITS>,
+	<C as CurveAffine>::Base: FieldExt,
+	<C as CurveAffine>::ScalarExt: FieldExt,
+{
+	/// Returns a new `AssignedPoint` given its coordinates as `AssignedInteger`
+	pub fn new(
+		x: AssignedInteger<C::Base, N, NUM_LIMBS, NUM_BITS, P>,
+		y: AssignedInteger<C::Base, N, NUM_LIMBS, NUM_BITS, P>,
+	) -> AssignedPoint<C, N, NUM_LIMBS, NUM_BITS, P> {
+		AssignedPoint { x, y }
+	}
+}
+
+/// Configuration elements for the circuit are defined here.
+#[derive(Debug, Clone)]
+pub struct EccAddConfig {
+	/// Constructs selectors from different circuits.
+	integer_reduce_selector: Selector,
+	integer_sub_selector: Selector,
+	integer_mul_selector: Selector,
+	integer_div_selector: Selector,
+}
```

Comment:
We should move this and every other config inside ecc/mod.rs file and use it for both same_curve and generic